src/server/apollo/dataSources.ts

```diff
@@ -180,7 +182,10 @@ export class CustomAPI extends RESTDataSource {
     if (!cacheTimestamp || cacheTimestamp < modifiedAt) {
       // Check if referenceData has changed. In this case, refresh choices instead of using cached ones.
       const body = { query: this.processQuery(referenceData) };
-      const data = await this.post(url, { body });
+      let data = await this.post(url, { body });
+      if (typeof data === 'string') {
+        data = JSON.parse(data);
+      }
```

Comment:
This wasn't needed before, it is a bit annoying. I'm wondering if there hasn't been a change in RESTDataSource class with latest update that removed "didReceiveResponse" function. It seems that we were overriding it, but that it doesn't exist anymore on the base class, and I think it had the role of converting to JSON...